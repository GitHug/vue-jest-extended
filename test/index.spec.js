import VueJestExtended from '../src/index';

class FakeWrapper {
  constructor() {
    this.events = { };
  }

  emit(event, payload) {
    if (this.events[event]) {
      this.events[event].push([payload]);
    } else {
      Object.assign(this.events, { [event]: [[payload]] });
    }
  }

  emitted(event) {
    return event ? this.events[event] : this.events;
  }
}

expect.extend(VueJestExtended);

describe('VueJestExtended', () => {
  describe('.toHaveEmitted', () => {
    it('passes when one of the events emitted by the wrapper matches the event given', () => {
      const wrapper = new FakeWrapper();
      wrapper.emit('input', { data: 'test' });
      wrapper.emit('change', { test: 'data' });

      expect(wrapper).toHaveEmitted('input');
    });

    it('fails if none of the events emitted by the wrapper matches the event given', () => {
      const wrapper = new FakeWrapper();
      wrapper.emit('change', { data: 'test' });

      expect(() => {
        expect(wrapper).toHaveEmitted('input');
      }).toThrowErrorMatchingSnapshot();
    });

    it('fails if no event has been emitted', () => {
      const wrapper = new FakeWrapper();

      expect(() => {
        expect(wrapper).toHaveEmitted('input');
      }).toThrowErrorMatchingSnapshot();
    });

    describe('.not.toHaveEmitted', () => {
      it('fails when one of the events emitted by the wrapper matches the event given', () => {
        const wrapper = new FakeWrapper();
        wrapper.emit('input', { data: 'test' });

        expect(() => {
          expect(wrapper).not.toHaveEmitted('input');
        }).toThrowErrorMatchingSnapshot();
      });

      it('passes if none of the events emitted by the wrapper matches the event given', () => {
        const wrapper = new FakeWrapper();
        wrapper.emit('change', { data: 'test' });

        expect(wrapper).not.toHaveEmitted('input');
      });

      it('passes if no event has been emitted', () => {
        const wrapper = new FakeWrapper();

        expect(wrapper).not.toHaveEmitted('input');
      });
    });
  });

  describe('.toHaveEmittedPayload', () => {
    it('passes when the payload of the event emitted by the wrapper matches the payload given', () => {
      const wrapper = new FakeWrapper();
      wrapper.emit('input', { data: 'test' });
      wrapper.emit('change', { test: 'data' });

      expect(wrapper).toHaveEmittedPayload('input', { data: 'test' });
    });

    it('fails if the payload of the event does not match the payload given', () => {
      const wrapper = new FakeWrapper();
      wrapper.emit('input', { data: 'test' });
      wrapper.emit('change', { test: 'data' });

      expect(() => {
        expect(wrapper).toHaveEmittedPayload('input', { test: 'data' });
      }).toThrowErrorMatchingSnapshot();
    });

    it('fails if the event has not been emitted', () => {
      const wrapper = new FakeWrapper();
      wrapper.emit('change', { test: 'data' });

      expect(() => {
        expect(wrapper).toHaveEmittedPayload('input', { test: 'data' });
      }).toThrowErrorMatchingSnapshot();
    });

    it('fails if no event has been emitted', () => {
      const wrapper = new FakeWrapper();

      expect(() => {
        expect(wrapper).toHaveEmittedPayload('input', { test: 'data' });
      }).toThrowErrorMatchingSnapshot();
    });

    describe('.not.toHaveEmittedPayload', () => {
      it('fails when the payload of the event emitted by the wrapper matches the payload given', () => {
        const wrapper = new FakeWrapper();
        wrapper.emit('input', { data: 'test' });
        wrapper.emit('change', { test: 'data' });

        expect(() => {
          expect(wrapper).not.toHaveEmittedPayload('input', { data: 'test' });
        }).toThrowErrorMatchingSnapshot();
      });

      it('passes if the payload of the event does not match the payload given', () => {
        const wrapper = new FakeWrapper();
        wrapper.emit('input', { data: 'test' });
        wrapper.emit('change', { test: 'data' });

        expect(wrapper).not.toHaveEmittedPayload('input', { test: 'data' });
      });

      it('passes if the event has not been emitted', () => {
        const wrapper = new FakeWrapper();
        wrapper.emit('change', { test: 'data' });

        expect(wrapper).not.toHaveEmittedPayload('input', { test: 'data' });
      });

      it('passes if no event has been emitted', () => {
        const wrapper = new FakeWrapper();

        expect(wrapper).not.toHaveEmittedPayload('input', { test: 'data' });
      });
    });
  });
});
