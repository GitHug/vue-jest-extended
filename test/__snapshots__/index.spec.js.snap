// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueJestExtended .toHaveEmitted .not.toHaveEmitted fails when one of the events emitted by the wrapper matches the event given 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toHaveEmitted([22m[32mexpected[39m[2m)[22m


        Expected wrapper not to have emitted:

          [32m\\"input\\"[39m
        Received:

          [31m{\\"input\\": [{\\"data\\": \\"test\\"}]}[39m
        "
`;

exports[`VueJestExtended .toHaveEmitted fails if no event has been emitted 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


        Expected wrapper to have emitted:

          [32m\\"input\\"[39m
        Received:

          [31m{}[39m
        "
`;

exports[`VueJestExtended .toHaveEmitted fails if none of the events emitted by the wrapper matches the event given 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


        Expected wrapper to have emitted:

          [32m\\"input\\"[39m
        Received:

          [31m{\\"change\\": [{\\"data\\": \\"test\\"}]}[39m
        "
`;

exports[`VueJestExtended .toHaveEmittedPayload .not.toHaveEmittedPayload fails when the payload of the event emitted by the wrapper matches the payload given 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toHaveEmitted([22m[32mexpected[39m[2m)[22m


        Expected wrapper not to have emitted:

          [32m{\\"input\\": [{\\"data\\": \\"test\\"}]}[39m
        Received:

          [31m{\\"change\\": [{\\"test\\": \\"data\\"}], \\"input\\": [{\\"data\\": \\"test\\"}]}[39m
        "
`;

exports[`VueJestExtended .toHaveEmittedPayload fails if no event has been emitted 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


        Expected wrapper to have emitted:

          [32m{\\"input\\": [{\\"test\\": \\"data\\"}]}[39m
        Received:

          [31m{}[39m
        "
`;

exports[`VueJestExtended .toHaveEmittedPayload fails if the event has not been emitted 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


        Expected wrapper to have emitted:

          [32m{\\"input\\": [{\\"test\\": \\"data\\"}]}[39m
        Received:

          [31m{\\"change\\": [{\\"test\\": \\"data\\"}]}[39m
        "
`;

exports[`VueJestExtended .toHaveEmittedPayload fails if the payload of the event does not match the payload given 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


        Expected wrapper to have emitted:

          [32m{\\"input\\": [{\\"test\\": \\"data\\"}]}[39m
        Received:

          [31m{\\"change\\": [{\\"test\\": \\"data\\"}], \\"input\\": [{\\"data\\": \\"test\\"}]}[39m
        "
`;
